src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -272,6 +273,14 @@ def __eq__(self, other: "SnowflakePlan") -> bool:
     def __hash__(self) -> int:
         return hash(self._id) if self._id else super().__hash__()
 
+    @property
+    def uuid(self) -> str:
+        return self._uuid
+
+    @uuid.setter
+    def uuid(self, value: str) -> None:
```

Comment:
i don't think we should provide a setter function here, i don't think anyone is suppose to change this once produced.